account_payment_order/models/account_payment_order.py

```diff
@@ -0,0 +1,444 @@
+                record.allowed_journal_ids = record.payment_mode_id.fixed_journal_id
+            elif record.payment_mode_id.bank_account_link == "variable":
+                record.allowed_journal_ids = record.payment_mode_id.variable_journal_ids
+            else:
+                record.allowed_journal_ids = False
+
+    def unlink(self):
+        for order in self:
+            if order.state == "uploaded":
+                raise UserError(
+                    _(
+                        "You cannot delete an uploaded payment order. You can "
+                        "cancel it in order to do so."
+                    )
+                )
+        return super(AccountPaymentOrder, self).unlink()
+
+    @api.constrains("payment_type", "payment_mode_id")
+    def payment_order_constraints(self):
+        for order in self:
+            if (
+                order.payment_mode_id.payment_type
+                and order.payment_mode_id.payment_type != order.payment_type
+            ):
+                raise ValidationError(
+                    _(
+                        "The payment type (%(ptype)s) is not the same as the payment "
+                        "type of the payment mode (%(pmode)s)",
+                        ptype=order.payment_type,
+                        pmode=order.payment_mode_id.payment_type,
+                    )
+                )
+
+    @api.constrains("date_scheduled")
+    def check_date_scheduled(self):
+        today = fields.Date.context_today(self)
+        for order in self:
+            if order.date_scheduled:
+                if order.date_scheduled < today:
+                    raise ValidationError(
+                        _(
+                            "On payment order %(porder)s, the Payment Execution Date "
+                            "is in the past (%(exedate)s).",
+                            porder=order.name,
+                            exedate=order.date_scheduled,
+                        )
+                    )
+
+    @api.depends("payment_line_ids", "payment_line_ids.amount_company_currency")
+    def _compute_total(self):
+        for rec in self:
+            rec.total_company_currency = sum(
+                rec.mapped("payment_line_ids.amount_company_currency") or [0.0]
+            )
+
+    @api.depends("payment_ids")
+    def _compute_payment_count(self):
+        for order in self:
+            order.payment_count = len(order.payment_ids)
+
+    @api.depends("move_ids")
+    def _compute_move_count(self):
+        rg_res = self.env["account.move"].read_group(
+            [("payment_order_id", "in", self.ids)],
+            ["payment_order_id"],
+            ["payment_order_id"],
+        )
+        mapped_data = {
+            x["payment_order_id"][0]: x["payment_order_id_count"] for x in rg_res
+        }
+        for order in self:
+            order.move_count = mapped_data.get(order.id, 0)
+
+    @api.model
+    def create(self, vals):
```

Comment:
I just tried to install the module locally and I notice the following warnings:

```
2023-02-08 14:15:52,874 1088979 WARNING db123 odoo.api.create: The model odoo.addons.account_payment_order.models.account_payment_order is not overriding the create method in batch 
2023-02-08 14:15:52,876 1088979 WARNING db123 odoo.api.create: The model odoo.addons.account_payment_order.models.account_payment_line is not overriding the create method in batch 

```

Could you change the create() method to support the creation in batch and make use of `@api.model_create_multi` ?

Reply:
This got corrected by [AnizR](https://github.com/AnizR) - merge PR into this PR